programs/zstdcli.c

```diff
@@ -1213,6 +1227,25 @@ int main(int const argCount, const char* argv[])
         if (adaptMin > cLevel) cLevel = adaptMin;
         if (adaptMax < cLevel) cLevel = adaptMax;
 
+        if (showDefaultCParams) {
+            size_t fileNb;
+            static const char* ZSTD_strategyMap[9 + 1] = { "", "ZSTD_fast",
+                "ZSTD_dfast", "ZSTD_greedy", "ZSTD_lazy", "ZSTD_lazy2", "ZSTD_btlazy2",
+                "ZSTD_btopt", "ZSTD_btultra", "ZSTD_btultra2"};
+            for (fileNb = 0; fileNb < (size_t)filenames->tableSize; fileNb++) {
+                const size_t fileSize = (size_t)UTIL_getFileSize(filenames->fileNames[fileNb]);
+                const size_t cParamsIdx = (fileSize < 16 KB) ? 3 : ( (fileSize < 128 KB) ? 2 : ( (fileSize < 256 KB) ? 1 : 0) );
```

Comment:
This way of doing embeds too much internal logic into `zstdcli.c` which is an external module (from a library perspective).
This is going to hurt maintenance.

Consider re-using a logic which is centrally defined.
In this case, there is `ZSTD_getCParams()` which is likely providing what you are looking for, without a need to expose and then access the internal tables directly.